src/Engine/ProtoCore/Lang/CallSite.cs

```diff
@@ -366,8 +395,197 @@ public void UpdateCallSite(int classScope, string methodName)
+                Dictionary<FunctionEndPoint, int> candidatesWithCastDistances =
+                    funcGroup.GetCastDistances(context, arguments, replicationControl.Instructions, runtimeCore.DSExecutable.classTable,
+                                               runtimeCore);
+
+                List<FunctionEndPoint> candidateFunctions = GetCandidateFunctions(stackFrame, candidatesWithDistances);
+                FunctionEndPoint compliantTarget = GetCompliantTarget(context, arguments,
+                                                                      replicationControl.Instructions, stackFrame, runtimeCore,
+                                                                      candidatesWithCastDistances, candidateFunctions,
+                                                                      candidatesWithDistances);
+
+                if (compliantTarget != null)
+                {
+                    return false; //Type conversion but no replication
+                }
+            }
+
+            #endregion
+
+            #region Case 5: Match with type conversion, replication and array promotion
+
+            {
+                //Build the possible ways in which we might replicate
+                replicationTrials =
+                    Replicator.BuildReplicationCombinations(replicationControl.Instructions, arguments, runtimeCore);
+
+                //Add as a first attempt a no-replication, but allowing up-promoting
+                replicationTrials.Insert(0,
+                                         new List<ReplicationInstruction>()
+                    );
+            }
+
+            #endregion
+
+            return true; //It'll replicate if it suceeds
+        }
+
+        /// <summary>
+        /// Call this method to obtain the Base64 encoded string that 
+        /// represent this instance of CallSite;s trace data
+        /// </summary>
+        /// <returns>Returns the Base64 encoded string that represents the
+        /// trace data of this callsite
+        /// </returns>
+        /// 
+        public string GetTraceDataToSave()
+        {
+            //Test to see if there is any actual data in the trace cache
+            if (!this.traceData.Any(srtd => srtd.HasAnyNestedData))
+                return null;
+
+            TraceSerialiserHelper helper = new TraceSerialiserHelper();
+            helper.TraceData = this.traceData;
+
+            using (MemoryStream memoryStream = new MemoryStream())
+            {
+
+                IFormatter formatter = new SoapFormatter();
+                formatter.Serialize(memoryStream, helper);
+
+                return Convert.ToBase64String(memoryStream.ToArray());
+            }
+
+        }
+
+        /// <summary>
+        /// Get all serializables that were created historically, but
+        /// were not re-created in the most recent graph update.
+        /// </summary>
+        public IList<ISerializable> GetOrphanedSerializables()
```

Comment:
This is a new method. 

This is the primary reconciliation method for the `CallSite`. This should only run all the way through the first time after the `CallSite` is created. On future runs, there won't be any `historicalSerializables`, so it'll bail early.